subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -1041,13 +1043,13 @@ uint8_t ll_df_set_conn_cte_tx_params(uint16_t handle, uint8_t cte_types, uint8_t
 	df_tx_cfg->ant_sw_len = switch_pattern_len;
 
 	df_tx_cfg->cte_types_allowed = cte_types;
-	df_tx_cfg->state = DF_CTE_CONN_TX_PARAMS_SET;
+	df_tx_cfg->is_initialized = true;
```

Comment:
Is it always typesafe (regardless of compiler) to assign `true` and `false` to a `uint8_t:1` field?

Reply:
It is if a field is bool:1. I've changed the type to `bool` to state that its purpose it to be a flag.

Reply:
But didn't we have that discussion in your earlier PR? 

e.g. if we have
```
bool:1 b_val;
uint8_t:7 u_val;
```
Is that always 1 octet, or will it be 2 octets, and is it the same for all compilers? 

Personally I think that, to be type safe, we should assign `1U` and `0U` if it is `uintX_t:1`, and `true` and `false` if it is `bool:1`, but `bool:1` should perhaps only be used if it can be used together with `uintX_t` bitfields, or if there are no more free bits in the bitfield.

Reply:
> But didn't we have that discussion in your earlier PR?

That is very possible. I've taken a look on the topic again and found better understanding.

Based on cppref (but in reference to C language ;-)): https://en.cppreference.com/w/c/language/bit_field

> Is that always 1 octet, or will it be 2 octets, and is it the same for all compilers?

`Declares a member with explicit width, in bits. Adjacent bit field members may be packed to share and straddle the individual bytes.` and `Multiple adjacent bit fields are permitted to be (and usually are) packed together`
This is related with all types.

What more: `Even though the number of bits in the object representation of _Bool is at least CHAR_BIT, the width of the bit field of type _Bool cannot be greater than1 (since C99).`

Summarizing, there is no guarantee that any bitfield will be packed together, even based on the same type. 
In case of gcc and clang in most of recent revisions merge bool with uint8_t fields.

```
struct some {
    bool a:1;
    uint16_t b:6;
    bool d:1;
    uint16_t c:1;
};
``` 
This struct has size of 2 when compiled with x86-gcc-11.2. Also this one has size of 2:

```
struct some {
    uint8_t a:1;
    uint16_t b:6;
    uint8_t d:1;
    uint16_t c:1;
};
```

Fortunately this is not a structure that isn't used to store PDUs data.

Reply:
I've decided to change `bool` to `uint8_t` and assignments to these members to `1U/0U`. I'll stick to convention that boolean members in bitfields are also defined as `uint8_t` unless all members are booleans.